src/components/plots/linePlot.ts

```diff
@@ -154,33 +154,70 @@ export module Plot {
       return { data: data, pixelPoints: pixelPoints, selection: d3.selectAll(allElements) };
     }
 
+    /**
+     * Retrieves the closest PlotData to queryPoint.
+     *
+     * Lines implement an x-dominant notion of distance; points closest in x are
+     * tie-broken by y distance.
+     *
+     * @param {Point} queryPoint The point to which plot data should be compared
+     *
+     * @returns {PlotData} The PlotData closest to queryPoint
+     */
     public getClosestPlotData(queryPoint: Point): PlotData {
-      var closestDistanceSquared = Infinity;
-      var closestDatum: any;
-      var closestSelection: D3.Selection;
-      var closestPoint: Point;
+      var minXDist = Infinity;
+      var minYDist = Infinity;
+
+      var closest: {
+        datum: any;
+        pixelPoint: Point;
+        node: Node
+      }[];
+
+      this.datasetOrder().forEach((key: string) => {
+        var plotData = this.getAllPlotData(key);
+        plotData.pixelPoints.forEach((pxPt: Point, index: number) => {
+          if (pxPt.x < 0 || pxPt.y < 0 || pxPt.x > this.width() || pxPt.y > this.height()) {
+            return;
+          }
+
+          var xDist = Math.abs(queryPoint.x - pxPt.x);
+          var yDist = Math.abs(queryPoint.y - pxPt.y);
 
-      this.datasetOrder().forEach((datasetKey: string) => {
-        var plotData = this.getAllPlotData(datasetKey);
-        plotData.pixelPoints.forEach((pixelPoint: Point, index: number) => {
-          var pixelPointDist = _Util.Methods.distanceSquared(queryPoint, pixelPoint);
-          if (pixelPointDist < closestDistanceSquared) {
-            closestDistanceSquared = pixelPointDist;
-            closestDatum = plotData.data[index];
-            closestPoint = pixelPoint;
-            closestSelection = plotData.selection;
+          if (xDist < minXDist || xDist === minXDist && yDist < minYDist) {
+            closest = [];
+            minXDist = xDist;
+            minYDist = yDist;
+          }
+
+          if (xDist === minXDist && yDist === minYDist) {
+            closest.push({
+              datum: plotData.data[index],
+              pixelPoint: pxPt,
+              node: plotData.selection[0][0]
+            });
           }
         });
       });
 
-      if (closestDatum == null) {
+      if (minXDist === Infinity) {
         return { data: [], pixelPoints: [], selection: d3.select() };
       }
 
+      var data: any[] = [];
+      var pixelPoints: Point[] = [];
+      var nodes: Node[] = [];
+
+      closest.forEach((c: any) => {
+        data.push(c.datum);
+        pixelPoints.push(c.pixelPoint);
+        nodes.push(c.node);
+      });
```

Comment:
Instead of playing with these closest objects, why not deal directly with data, pixelPoints, and nodes.  It should cut down quite a bit on number of lines.